src/secrets/CommandEdit.ts

```diff
@@ -54,41 +55,85 @@ class CommandEdit extends CommandPolykey {
+            }
+            return exists;
+          },
+          meta,
+        );
+        // If the editor exited with a code other than zero, then execSync
+        // will throw an error. So, in the case of saving the file but the
+        // editor crashing, the program won't save the updated secret.
+        execSync(`${process.env.EDITOR} \"${tmpFile}\"`, { stdio: 'inherit' });
+        let content: string;
         try {
-          content = await this.fs.promises.readFile(tmpFile);
+          content = (await this.fs.promises.readFile(tmpFile)).toString(
+            'binary',
+          );
         } catch (e) {
-          throw new errors.ErrorPolykeyCLIFileRead(e.message, {
-            data: {
-              errno: e.errno,
-              syscall: e.syscall,
-              code: e.code,
-              path: e.path,
-            },
-            cause: e,
-          });
+          // If the secret exists but the file doesn't, then something went
+          // wrong, and the file cannot be read anymore. This is bad.
+          // If the secret didn't exist before and we can't read the file,
+          // then the secret was never actually created or saved. The user
+          // doesn't want to make the secret anymore, so abort mision!
+          if (e.code === 'ENOENT' && secretExists) {
+            throw new errors.ErrorPolykeyCLIFileRead(e.message, {
+              data: {
+                errno: e.errno,
+                syscall: e.syscall,
+                code: e.code,
+                path: e.path,
+              },
+              cause: e,
+            });
+          }
+          return;
         }
-        await pkClient.rpcClient.methods.vaultsSecretsEdit({
-          nameOrId: secretPath[0],
-          secretName: secretPath[1],
-          secretContent: content.toString('binary'),
-        });
+        await binUtils.retryAuthentication(async (auth) => {
+          // This point will never be reached if the secret exists but the temp
+          // file doesn't. As such, if the file doesn't exist, it can only mean
+          // that the secret didn't exist before, and the user didn't write to
+          // the file, either.
+          // Otherwise, the file exists, and by extension, so does the secret.
+          // In this case, update the secret instead of making a new one.
+          if (!secretExists && fs.existsSync(tmpFile)) {
+            await pkClient.rpcClient.methods.vaultsSecretsNew({
+              metadata: auth,
+              nameOrId: secretPath[0],
+              secretName: secretPath[1],
+              secretContent: content,
+            });
+          } else {
+            await pkClient.rpcClient.methods.vaultsSecretsEdit({
+              metadata: auth,
+              nameOrId: secretPath[0],
+              secretName: secretPath[1],
+              secretContent: content,
+            });
+          }
+        }, meta);
         await this.fs.promises.rm(tmpDir, { recursive: true, force: true });
```

Comment:
This can leak the secret if we run into any problems before this point. You need to remove the temp file in the finally block just below to ensure that it's always removed. You also need to handle any `ENOENT` errors when trying this in case it directory you want to remove doesn't exist anymore. Any other error should still be thrown.